src/App.jsx

```diff
@@ -1,24 +1,83 @@
 import AddTodoForm from './AddTodoForm';
 import TodoList from './TodoList';
-import { useSemiPersistentState } from './useSemiPersistentState';
+import React, { useState, useEffect } from 'react';
+import { BrowserRouter, Routes, Route } from 'react-router-dom';
 
 function App() {
-  const [todoList, setTodoList] = useSemiPersistentState();
-  
+  const [todoList, setTodoList] = useState(() => {
+    const savedTodos = localStorage.getItem("savedTodoList"); // Load saved todos
+    return savedTodos ? JSON.parse(savedTodos) : []; // Default to empty array
+  });
```

Comment:
you dont need to fetch from localstorage again since we are fetching from an api initialise with an empty array


```suggestion
  const [todoList, setTodoList] = useState([])
```